Check code format correctness in Travis CI

skalee opened the pull request:
As of PR #6, the Rust code is formatted with Rustfmt tool.  Let's add Rustfmt checks in Travis CI to ensure that code format correctness is not violated.

skalee committed:
Inline default Travis CI script for Rust

According to docs, Travis CI default build script is like that.  Let's
state it explicitly before modifying it.

See: https://docs.travis-ci.com/user/languages/rust

ronaldtse commented:
@skalee thanks, while you're at it, could you help migrate this to GitHub Actions? It's a lot faster for tests. We've been migrating all our other projects too!

skalee committed:
Check code format in Travis CI

As of PR #6, the Rust code is formatted with Rustfmt tool.  Let's add
Rustfmt checks in Travis CI to ensure that code format correctness is
not violated.

See: https://github.com/rust-lang/rustfmt

skalee commented:
Yes, probably. I don't know how they work though. I got to do some research.

ronaldtse commented:
@skalee It allows running of Ubuntu, macOS and Windows. It also allows running Docker containers.

* Sample for 3 platforms: https://github.com/metanorma/metanorma-csand/pull/18
* Sample for Docker container: https://github.com/metanorma/mn-samples-iso/blob/master/.github/workflows/publish.yml

skalee committed:
wip test this way

skalee commented:
@ronaldtse Does it mean that unit tests should be migrated from Travis to GH Actions as well?

ronaldtse commented:
@skalee yes, we're going to abandon Travis. It was one of @dewyatt 's dream.

skalee commented:
I'm not astonished. Using Docker in Travis is quite unpleasant.

Opened a new issue #10.

skalee commented:
Closing. It was re-implemented in #12.
